test/geometry/liegroup/test_so3.py

```diff
@@ -124,9 +144,24 @@ def test_matrix(self, device, dtype, batch_size):
             qp_ = q1 * pquat * q1.inv()
             rp_ = torch.matmul(r1, pvec)[None, :]
             self.assert_close(rp_, qp_.vec)  # p_ = R*p = q*p*q_inv
+            self.assert_close(rp_.norm(), pvec.norm())
+
+    @pytest.mark.parametrize("batch_size", (1, 2, 5))
+    def test_ortho(self, device, dtype, batch_size):
+        q = Quaternion.random(batch_size)
```

Comment:
@cjpurackal wondering whether we should make batch_size as optional and default value to None indicating unhatched data

Reply:
unhatched data as in raw tensors?

Reply:
now a default batch size one will make a (1,3) which is compatible to the rest of liegroup package. My proposal with batch size None will make (3,) data which is not really tested, specially where we use @ for batched ops

Reply:
I think that's a good idea

Reply:
I believe also we need to take care of this when we do from_coefss